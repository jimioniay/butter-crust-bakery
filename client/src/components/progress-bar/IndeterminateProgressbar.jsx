import React from 'react';

import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import './progress-bar.scss';

const IndeterminateProgressBar = ({ status }) => {
  if (typeof status === 'object' ? status.status : status) {
    return (
      <div className="progress">
        <div className="indeterminate" />
      </div>
    );
  }
  return null;
};

const mapStateToProps = ({
  postTransferClientReducer: { spinner },
  loadingReducer: { load },
}) => ({
  status: spinner || load,
});

export default connect(
  mapStateToProps,
  {},
)(IndeterminateProgressBar);

IndeterminateProgressBar.defaultProps = {
  status: false,
};

IndeterminateProgressBar.propTypes = {
  status: PropTypes.bool,
};
