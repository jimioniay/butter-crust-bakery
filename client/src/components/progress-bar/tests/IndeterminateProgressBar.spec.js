import React from 'react';
import IndeterminateProgressbar from '../IndeterminateProgressbar';

describe('Indeterminate progress bar', () => {
  const wrapper = shallow(<IndeterminateProgressbar loading />);

  it('renders properly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
