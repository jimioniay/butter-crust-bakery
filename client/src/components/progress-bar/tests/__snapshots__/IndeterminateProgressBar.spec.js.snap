// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Indeterminate progress bar renders properly 1`] = `ShallowWrapper {}`;
